orchagent/txmonitororch.cpp

```diff
@@ -0,0 +1,367 @@
+    m_monitorStateTable.set(key, fieldValuesVector);
+}
+
+
+void TxMonitorOrch::updateParams(const string& key, const vector<FieldValueTuple>& data)
+{
+    SWSS_LOG_ENTER();
+
+    if(key == KEY_PoolingPeriod)
+    {
+        setTimer(data);
+    }
+    else if (key == KEY_Threshold)
+    {
+        setThreshold(data);
+    }
+    else
+    {
+        SWSS_LOG_ERROR("Unknown key %s", key.c_str());
+    }
+}
+
+
+
+void TxMonitorOrch::mapPortToName()
+{
+    SWSS_LOG_ENTER();
+
+    for (auto const &curr : gPortsOrch->getAllPorts())
+    {
+        string portName = curr.first;
+        Port port = curr.second;
+        string portOid;
+
+        if (port.m_type != Port::Type::PHY)
+        {
+            continue;
+        }
+
+        if (!m_countersPortNameMap->hget("", portName, portOid))
+        {
+            SWSS_LOG_ERROR("error getting port name from counters");
+            continue;
+        }
+        m_portsStringsMap[port.m_alias] = portOid;
+    }
+}
+
+void TxMonitorOrch::initReadCounter()
+{
+    SWSS_LOG_ENTER();
+
+    for (auto &entry : m_portsStringsMap)
+    {
+        string portAlias = entry.first;
+        string portOid = entry.second;
+        string portState = "OK";
+        string strVal;
+
+        if (!m_countersTable->hget(portOid, counterName, strVal))
+        {
+            SWSS_LOG_WARN("Error reading counters table");
+            SWSS_LOG_ERROR("Cannot take information from table for port %s", portOid.c_str());
+            strVal = "0";
+            portState = "UNKNOWN";
+        }
+
+        int newTxCounter;
+
+        try
+        {
+            newTxCounter = stoi(strVal);
+
+            if(newTxCounter < 0)
+            {
+                throw logic_error("Non-logical tx-counter value");
+            }
+        }
+        catch (...)
+        {
+            newTxCounter = 0;
+            portState = "UNKNOWN";
+
+            SWSS_LOG_ERROR("Tx-counter value invalid");
+        }
+
+        m_portsMap_txCounter[portAlias] = newTxCounter;
+        insertInfoToDbState(portAlias, portState);
+    }
+    m_monitorStateTable.flush();
+}
+
+
+
+void TxMonitorOrch::initTimer()
+{
+    SWSS_LOG_ENTER();
+
+    auto interv = timespec { .tv_sec = (int)m_pooling_period, .tv_nsec = 0 };
+    m_timer = new SelectableTimer(interv);
+    auto executor = new ExecutableTimer(m_timer, this, "Tx_Port_Monitor");
+    Orch::addExecutor(executor);
+    m_timer->start();
+}
+
+void TxMonitorOrch::setTimer(const vector<FieldValueTuple>& data)
+{
+    SWSS_LOG_ENTER();
+
+    int new_PoolingPeriod;
+
+    for (auto i : data)
+    {
+        new_PoolingPeriod = Default_PoolingPeriod;
+        const auto &field = fvField(i);
+        const auto &value = fvValue(i);
+
+        if (field == FIELD_VALUE)
+        {
+            try
+            {
+                new_PoolingPeriod = stoi(value);
+                if(new_PoolingPeriod <= 0)
+                {
+                    throw invalid_argument("Illegal pooling period");
```

Comment:
You can use SWSS_LOG_THROW instead of this throw and remove print to log in the catch part.